lib/templates/lab2/lab2.dart

```diff
@@ -1,19 +1,81 @@
-class Book {
-  // your code here
+
+
+void main() {
+  FictionBook fictionBook = FictionBook('Lean Startup', 'Eric Ries', 366, 'Non-fiction');
+  //var book1 = Book('Lean Startup', 'Eric Ries', 366);
+  fictionBook.bookDetails();
+  print('\n');
+  print(fictionBook.toJson());
+
 }
 
-class FictionBook {
-  // your code here
+class Book with JsonSerializable  {
+  String title;
+  String author;
+  int numberOfPages;
+  
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails() {
+    print('Title: $title');
+    print('Author: $author');
+    print('Number of pages: $numberOfPages');
+  }
+}
+
+class FictionBook extends Book{
+  String genre;
+
+  FictionBook(super.title, super.author, super.numberOfPages, this.genre);
+  
+  @override 
+  void bookDetails() {
+    super.bookDetails();
+    print('Genre: $genre');
+  }
 }
 
 mixin JsonSerializable {
-  // your code here
+
+  String get title;
+  String get author;
+  int get numberOfPages;
+  
+  Map<String, dynamic> toJson() => {
+    'title' : title,
+    'author': author,
+    'numberofPages': numberOfPages,
+  };
 }
 
-double calculateArea(){
-  // your code here
+double calculateArea(double length, double width, String shape){
+  //Rectangle
+  double area;
+  if(shape == 'circle') {
+    area = 3.14 * length * length;
```

Comment:
where is default rectange?

Reply:
_an optional parameter shape of type String with a default value of "rectangle"_

As I understand it is optional, isn't it?